kart/point_cloud/dataset1.py

```diff
@@ -24,4 +32,29 @@ def __init__(self, tree, path, dirname=None, repo=None):
 
         self.L = logging.getLogger(self.__class__.__qualname__)
 
-    # TODO - lots more functionality.
+    @property
+    @functools.lru_cache(maxsize=1)
+    def tiles_tree(self):
+        """Returns the root of the tiles tree, or the empty tree if no tiles tree exists."""
+        if self.inner_tree:
+            try:
+                return self.inner_tree / self.TILES_PATH
+            except KeyError:
+                pass
+        return self.repo.empty_tree if self.repo else None
```

Comment:
is `self.repo is None` a valid state? how does that happen? I'm not clear that returning None here is a sensible thing to do. Similarly unclear on whether returning an empty tree is helpful. Maybe this needs more comments explaining those states

Reply:
Yeah the dataset constructor interface is a little confused - a symptom of the long evolution of datasets up to this point, and the variety of different ways they are now used. Every dataset currently has the same constructor, which is used as an interface, and most constructors still handle every special case (partly because I think it will end up in a common-to-all-datasets base class soon).
This particular weirdness (repo maybe None but maybe isn't) I think is a symptom of the following causes:
- I had a bug "Fix for feature-KeyError in empty dataset" and I fixed it by making methods that return trees return empty-trees if they had no data (ie, dataset doesn't exist, dataset does exist but doesn't have a `feature` directory, etc). That means no dataset needs to be special cased - `.feature_tree` always returns a valid tree.
- Returning an empty-tree means wiring in the repo, which a dataset doesn't otherwise depend directly. I did this but made it optional (which was a bit messy and maybe the wrong call)...
- which means that unit tests on the Dataset3 class which don't even use a git repo still work - they just create some tree-like objects that support the `/` operator, but they don't create a repo-like object or a global empty-tree object.

Anyway you are right it needs cleaning up but I think for now I'll just make this particular dataset require a non-null repo, and I'll fix the rest of it when I figure out what the common base-class of all datasets should be.